payments/providers/bambora_payform.py

```diff
@@ -0,0 +1,317 @@
+        }
+
+    def initiate_payment(self, order: Order) -> str:
+        """Initiate payment by constructing the payload with necessary items"""
+
+        payload = {
+            "version": "w3.1",
+            "api_key": self.config.get(VENE_PAYMENTS_BAMBORA_API_KEY),
+            "payment_method": {
+                "type": "e-payment",
+                "return_url": self.get_success_url(),
+                "notify_url": self.get_notify_url(),
+                "selected": self.config.get(VENE_PAYMENTS_BAMBORA_PAYMENT_METHODS),
+                "token_valid_until": datetime.combine(
+                    order.due_date, datetime.max.time()
+                ).timestamp(),
+            },
+            "currency": "EUR",
+            "order_number": str(order.order_number),
+        }
+
+        self.payload_add_products(payload, order)
+        self.payload_add_customer(payload, order)
+        self.payload_add_auth_code(payload)
+
+        try:
+            r = requests.post(self.url_payment_auth, json=payload, timeout=60)
+            r.raise_for_status()
+            return self.handle_initiate_payment(order, r.json())
+        except RequestException as e:
+            raise ServiceUnavailableError(_("Payment service is unreachable")) from e
+
+    def handle_initiate_payment(self, order: Order, response) -> str:
+        """Handling the Bambora payment auth response"""
+        result = response["result"]
+        if order.status == OrderStatus.EXPIRED:
+            raise ExpiredOrderError(_("Order has already expired"))
+        if result == 0:
+            # Create the URL where user is redirected to complete the payment
+            # Append "?minified" to get a stripped version of the payment page
+            return self.url_payment_token.format(token=response["token"])
+        elif result == 1:
+            raise PayloadValidationError(
+                f"{_('Payment payload data validation failed: ')} {' '.join(response['errors'])}"
+            )
+        elif result == 2:
+            raise DuplicateOrderError(_("Order with the same ID already exists"))
+        elif result == 10:
+            raise ServiceUnavailableError(_("Payment service is down for maintenance"))
+        else:
+            raise UnknownReturnCodeError(
+                f"{_('Return code was not recognized: ')} {result}"
+            )
+
+    def payload_add_products(self, payload: dict, order: Order):
+        """Attach info of bought products to payload
+
+        Order lines that contain bought products are retrieved through order"""
+        order_lines: [OrderLine] = OrderLine.objects.filter(order=order.id)
+        items: [dict] = []
+
+        product = order.product
+        int_tax = int(order.tax_percentage)
+        assert int_tax == product.tax_percentage  # make sure the tax is a whole number
+        items.append(
+            {
+                "id": str(product.id),
+                "title": product.name,
```

Comment:
do we need to activate some language for this?

Reply:
Somehow yes, but `product.name` calls a getter that does this, so I'm not sure how should we handle it.
```python
@property
def name(self):
    return f"{_('Berth product')}: {self}"

@property
def name(self):
    return f"{self.service} - {self.period}"
```

Reply:
maybe we should do something like:
```python
with activate(order.application.language):
    product_name = product.name
```
🤔

Reply:
Done :D